File: WhichAreIn.cpp
// https://www.codewars.com/kata/which-are-in/train/cpp
 
#include <vector>
#include <string>
#include <iostream>
#include <algorithm>

class WhichAreIn
{
	public:
	static std::vector<std::string> inArray(std::vector<std::string> &array1, std::vector<std::string> &array2) {
		std::vector<std::string> found;

		for (std::string ar2: array2) {
			for (std::string ar1: array1) {
				if ((ar2.find(ar1) != std::string::npos)) {
					found.push_back(ar1);
				}
			}
		}

		std::sort(found.begin(), found.end());
		found.erase(unique(found.begin(), found.end()), found.end());
		return found;
	}
};


int main() {
	std::vector<std::string> arr1 = { "arp", "live", "strong" };
	std::vector<std::string> arr2 = { "lively", "alive", "harp", "sharp", "armstrong" };
	for (std::string result: WhichAreIn::inArray(arr1, arr2)) {
		std::cout << result << std::endl;
	}
}

File: WhichAreIn.py
def in_array(array1, array2):
	found = set()
	for ar2 in array2:
		for ar1 in array1:
			if ar2.find(ar1) != -1:
				found.add(ar1)
	foundlist = list(found)
	foundlist.sort()
	return foundlist


arr1 = ["arp", "live", "strong"]
arr2 = ["lively", "alive", "harp", "sharp", "armstrong"]
print(in_array(arr1, arr2))
